src/test/java/de/caritas/cob/agencyservice/api/controller/AgencyControllerTest.java

```diff
@@ -0,0 +1,344 @@
+
+    when(agencyService.getAgenciesTopics())
+        .thenReturn(null);
+
+    mvc.perform(
+            get(PATH_GET_LIST_OF_AGENCIES_TOPICS)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isNoContent());
+  }
+
+  @Test
+  void getAgenciesTopics_Should_ReturnListAndOk_When_ServiceReturnsList() throws Exception {
+
+    List<AgencyTopicsDTO> agenciesTopics = new ArrayList<>();
+    agenciesTopics.add(AGENCY_TOPICS_DTO);
+
+    when(topicEnrichmentService.enrichTopicIdsWithTopicData(Mockito.anyList()))
+        .thenReturn(agenciesTopics);
+
+    mvc.perform(
+            get(PATH_GET_LIST_OF_AGENCIES_TOPICS))
+        .andExpect(status().isOk());
+
+    verify(topicEnrichmentService, atLeastOnce()).enrichTopicIdsWithTopicData(Mockito.anyList());
+  }
+
+  @Test
+  void getTenantAgencies_Should_ReturnNoContent_When_ServiceReturnsEmptyList() throws Exception {
+
+    when(agencyService.getAgencies(Mockito.anyString(), Mockito.anyInt()))
+        .thenReturn(null);
+
+    mvc.perform(
+            get(PATH_GET_LIST_OF_AGENCIES_PRIVATE + "?" + VALID_POSTCODE_QUERY + "&"
+                + VALID_TOPIC_ID_QUERY)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isNoContent());
+  }
+
+  @Test
+  void getTenantAgencies_Should_ReturnBadRequest_When_PostcodeParamIsInvalid() throws Exception {
+
+    mvc.perform(
+            get(PATH_GET_LIST_OF_AGENCIES_PRIVATE + "?" + INVALID_POSTCODE_QUERY + "&"
+                + VALID_TOPIC_ID_QUERY)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void getTenantAgencies_Should_ReturnBadRequest_When_topicIdParamIsNotProvided()
+      throws Exception {
+
+    mvc.perform(get(PATH_GET_LIST_OF_AGENCIES_PRIVATE + "?" + VALID_POSTCODE_QUERY)
+        .accept(MediaType.APPLICATION_JSON)).andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void getTenantAgencies_Should_ReturnBadRequest_When_PostCodeParamIsNotProvided()
+      throws Exception {
+
+    mvc.perform(get(PATH_GET_LIST_OF_AGENCIES_PRIVATE + "?" + VALID_TOPIC_ID_QUERY)
+        .accept(MediaType.APPLICATION_JSON)).andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void getTenantAgencies_Should_ReturnListAndOk_When_ServiceReturnsList() throws Exception {
+
+    List<FullAgencyResponseDTO> agencies = new ArrayList<>();
+    agencies.add(FULL_AGENCY_RESPONSE_DTO);
+
+    when(agencyService.getAgencies(Mockito.anyString(), Mockito.anyInt()))
+        .thenReturn(agencies);
+
+    mvc.perform(
+            get(PATH_GET_LIST_OF_AGENCIES_PRIVATE + "?" + VALID_POSTCODE_QUERY + "&"
+                + VALID_TOPIC_ID_QUERY)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("[0].name").value(AGENCY_RESPONSE_DTO.getName()));
+
+    verify(agencyService, atLeastOnce()).getAgencies(Mockito.anyString(), Mockito.anyInt());
```

Comment:
In your tests, you’re using Mockito.any(...) repeatedly. While this works, specifying argument matchers that align closely with the tested scenario is better if possible.

Reply:
I agree with you, however in this case 1st all the tests on this class where not added by me I only added the last tests that use PATH_GET_LIST_OF_AGENCIES_PRIVATE. But since I changed the test file name, now it shows that I did them all.
Having said that, I think, in this case we have to use the Mockito.any(...) because of the Optional.class used as parameter on the method. I could replace the Mockito.any for ArgumentMatchers.<Optional<String>>any(), however I dont think it is worth the effort, on this case.

Reply:
ok, this is fine